Capstone/View Controllers/HomeViewController.m

```diff
@@ -0,0 +1,102 @@
+//
+//  HomeViewController.m
+//  Capstone
+//
+//  Created by Abel Kelbessa on 7/4/22.
+//
+
+#import "HomeViewController.h"
+#import "HomeCell.h"
+#import "UIImageView+AFNetworking.h"
+#import "DetailsViewController.h"
+
+@interface HomeViewController ()<UITableViewDelegate, UITableViewDataSource>
+
+@property(strong, nonatomic) NSDictionary *allCollegeInfo;
+@property(strong, nonatomic) NSArray *colleges;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray *collegeNames;
+@property (strong, nonatomic) NSMutableArray *collegeCity;
+@property (strong, nonatomic) NSMutableArray *collegeDescription;
+@property (strong, nonatomic) NSMutableArray *collegeImage;
+
+@end
+
+@implementation HomeViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.collegeNames = [[NSMutableArray alloc]init];
+    self.collegeCity = [[NSMutableArray alloc]init];
+    self.collegeDescription = [[NSMutableArray alloc]init];
+    self.collegeImage = [[NSMutableArray alloc]init];
+    [self fetchData];
+    
+}
+
+// I will move this to the API manafer once I am done testing the filtering
+-(void)fetchData {
+    NSURL *url = [NSURL URLWithString:@"https://api.collegeai.com/v1/api/college-list?api_key=4c4e51cca8832178dcfb29217c&filters=%7B%0A%22funding-type%22%3A%5B%22public%22%5D%2C%0A%22schoolSize%22%3A%5B%22large%22%5D%2C%0A%22zipCode%22%3A%2202111%22%2C%0A%22distanceFromHomeMiles%22%3A%5B0%2C500%5D%2C%0A%22satOverall%22%3A1200%2C%0A%22closeToMyScores%22%3Atrue%0A%7D&info_ids=website%2CshortDescription%2ClongDescription%2CcampusImage%2Ccity%2CstateAbbr%2Caliases%2Ccolors%2ClocationLong%2ClocationLat"];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) {
+               NSLog(@"%@", [error localizedDescription]);
+           }
+           else {
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               self.allCollegeInfo = dataDictionary;
```

Comment:
Where else do you use self.allCollegeInfo?

Reply:
No where actually. I forgot to remove self.allCollegeInfo. My bad.